zcash_primitives/src/transaction/builder.rs

```diff
@@ -278,7 +278,7 @@ impl<'a, P: consensus::Parameters, R: RngCore> Builder<'a, P, R> {
     pub fn build(
         mut self,
         prover: &impl TxProver,
-    ) -> Result<(Transaction, SaplingMetadata), Error> {
+    ) -> Result<(Transaction, SaplingMetadata, Amount), Error> {
```

Comment:
This is a breaking API change, and it is also unnecessary because we know that the builder hard-codes `DEFAULT_FEE` which is part of the public crate API.

Revert this change; we should design the API changes for potentially-variable fee setting as a cohesive feature, separately from this PR.

Reply:
The "we know the builder hard-codes DEFAULT_FEE" is another one of those "implicit knowledge, you have to read the code or the docs to figure it out" things that I don't like having in our codebase. I'd like to keep this change.

Reply:
Fixed by https://github.com/zcash/librustzcash/compare/3a3827b559178ed8e9664b06c5f8425be0186d4c..6e228d35ecb33b605a3aece439d473c3c0424dd6